core/src/transport.rs

```diff
@@ -115,12 +108,20 @@ pub trait Transport {
     /// obtained from [dialing](Transport::dial).
     type Dial: Future<Output = Result<Self::Output, Self::Error>>;
 
+    fn poll(self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<TransportEvent<Self>>
```

Comment:
Decided to add `Transport::poll` instead of requiring the Transport to implement `Stream`. Reason for this is that a transport's stream should never terminate (never return `None`). This is also aligned with how most other structs in `rust-libp2p` do it. 
Q: Should we auto-implement `Stream` for `T: Transport`? Would ease polling if users use a Transport directly.

Reply:
> Reason for this is that a transport's stream should never terminate (never return `None`).

I think exposing a bare poll as the baseline is good. We shouldn't force users into a trait if not necessary.

> Q: Should we auto-implement `Stream` for `T: Transport`? Would ease polling if users use a Transport directly.

Yeah I would suggest we do. I might do the same for `StreamMuxer` over in https://github.com/libp2p/rust-libp2p/pull/2648 actually.

Reply:
Agreed with everything above. 

At some point it might make sense to drive the `InfiniteStream` effort upstream, or introduce our own `InfiniteStream` trait  in rust-libp2p.